recipe/meta.yaml

```diff
@@ -1,34 +1,41 @@
-{% set name = "typeguard" %}
 {% set version = "4.1.5" %}
+{% set min_python = "3.8" %}
+{% set min_typing_extensions = "4.7.0" %}
 
 package:
-  name: {{ name|lower }}
+  name: typeguard
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/t/typeguard/typeguard-{{ version }}.tar.gz
   sha256: ea0a113bbc111bcffc90789ebb215625c963411f7096a7e9062d4e4630c155fd
 
 build:
   noarch: python
-  number: 1
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  number: 2
+  script: {{ PYTHON }} -m pip install . --no-deps -vv --no-build-isolation
 
 requirements:
   host:
-    - python >=3.8
+    - python >={{ min_python }}
     - pip
     - setuptools_scm
   run:
-    - python >=3.8
     - importlib_metadata >=3.6
-    - typing_extensions >=4.7.0
+    - python >={{ min_python }}
+    - typing_extensions >={{ min_typing_extensions }}
   run_constrained:
     - pytest >=7
 
 test:
+  requires:
+    - pip
+    - python {{ min_python }}.*
+    - typing_extensions {{ min_typing_extensions }}.*
```

Comment:
force use of the minimum version

Reply:
How does that differ from what was done before? All I see is use of single-use variables without any semantic changes. And why was `typing_extensions` singled out?

Reply:
> single-use 

both are used twice: once to provide the lower pin for installation, once to pin _tightly_ during test so that `pip check` can catch incompatibilities.

>  And why was typing_extensions singled out?

See  https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/615

Reply:
> How does that differ from what was done before? All I see is use of single-use variables without any semantic changes.

Before, this package was *not tested* to see whether the conda dependency specs were compatible with the dependency specs listed in the typeguard pyproject.toml. By using pip check, builds of this recipe will fail if during the testing phase the packages installed by conda do not meet the requirements listed in the pyproject.toml.

Now, because the dependency versions in the test environment are strictly pinned to the same version that are listed in the requirements section, pip check will fail if the upstream pyproject toml has increased the minimum version for these dependencies.

> And why was typing_extensions singled out?

No reason, this recipe would be better if it checked both importlib_metadata and typing_extensions, but maybe the typing_extensions spec is updated more often and causes more issues for downstream packages.